silkworm/silkrpc/http/request_parser.cpp

```diff
@@ -24,270 +24,80 @@
-                return ResultType::indeterminate;
-            } else if (is_ctl(input)) {
-                return ResultType::bad;
-            } else {
-                req.headers.back().value.push_back(input);
-                return ResultType::indeterminate;
-            }
-        case expecting_newline_2:
-            if (input == '\n') {
-                state_ = header_line_start;
-                return ResultType::indeterminate;
-            } else {
-                return ResultType::bad;
-            }
-        case expecting_newline_3:
-            if (input == '\n') {
-                state_ = content_start;
-                // Look for Content-Length header to get content size
-                if (req.content_length == 0) {
-                    const auto it = std::find_if(req.headers.begin(), req.headers.end(), [&](const Header& h) {
-                        return h.name == "Content-Length";
-                    });
-                    if (it == req.headers.end()) {
-                        return ResultType::bad;
-                    }
-                    const char* str{it->value.c_str()};
-                    char* end_ptr{nullptr};
-                    errno = 0;
-                    long long len{std::strtoll(str, &end_ptr, 0)};
-                    if (errno == ERANGE || end_ptr == str || len < 0 || len > UINT32_MAX) {
-                        return ResultType::bad;
-                    }
-                    req.content_length = static_cast<uint32_t>(len);
-                }
-                if (req.content_length == 0) {
-                    return ResultType::good;
-                }
-                // Look for Expect header to handle continuation request
-                const auto it = std::find_if(req.headers.begin(), req.headers.end(), [&](const Header& h) {
-                    return h == kExpectRequestHeader;
-                });
-                if (it != req.headers.end()) {
-                    return ResultType::processing_continue;
-                }
-                return ResultType::indeterminate;
-            } else {
-                return ResultType::bad;
+    if (req.content_length != 0 && req.content.length() < req.content_length) {
+        for (size_t ii = 0; ii < slen; ii++) {
+            req.content.push_back(begin[ii]);
+        }
+        if (req.content.length() < req.content_length)
+            return ResultType::indeterminate;
+        else
+            return ResultType::good;
+    }
+
+    num_headers = sizeof(headers) / sizeof(headers[0]);
+    auto res = phr_parse_request(begin, slen, &method_name, &method_len, &path, &path_len, &minor_version, headers, &num_headers, last_len);
+    if (res < 0) {
+        return ResultType::bad;
+    }
+
+    bool expect_request = false;
+    bool content_length_present = false;
+
+    for (size_t ii = 0; ii < num_headers; ii++) {
+        if (memcmp(headers[ii].name, "Content-Length", headers[ii].name_len) == 0) {
```

Comment:
Either use `std::memcmp` here and below or add `using namespace std;` at the top

Reply:
ok